core/src/main/java/org/apache/druid/utils/JvmUtils.java

```diff
@@ -23,9 +23,25 @@
 
 import java.lang.management.ManagementFactory;
 import java.lang.management.ThreadMXBean;
+import java.util.StringTokenizer;
 
 public class JvmUtils
 {
+  private static final boolean IS_JAVA9_COMPATIBLE = isJava9Compatible(System.getProperty("java.specification.version"));
```

Comment:
A bit sad there isn't a better way to do this.

Reply:
Once Java 8 support is dropped, one can use https://docs.oracle.com/javase/9/docs/api/java/lang/Runtime.Version.html directly for version checks.